### DISCUSSION TITLE
NFT-DAO TL,DR Metadata

### DISCUSSION THREAD

**Discussion by DanM3rcurius** [NFT-DAO/NFT-Metadata#10] [Q&A]
Hello everyone,
I tried to scan our discussions, threads and content here for something like a summary or assessment on **if and what metadata can be stored in NFTs on chain**? I need this info in order to write the requirements for the use cases.

**Comment by stephen-rowan** [comment]
Hi Dan, I can't give you a definitive technical answer - but here are some pointers ...

The key overview is an IOHT Metadata workshop from January 2021 ([link](https://www.youtube.com/watch?v=LrN3ETZ3fRM)) and associated slides ([link](https://docs.google.com/presentation/d/1ursHchJiBP5ZVuXcW2uVJMmzXjlJk_di65CKmAplEy4/edit#slide=id.gb7121fc2c5_2_25)).

From a use case perspective some points of relevance might be :

1 - The *ultimate* limitation on what metadata can be stored onchain is size - from 200 bytes to a maximum of 16000 bytes (see Slide 7 of the IOHK presentation).
2 - However the more bytes that are stored onchain the greater the transaction fees - so there will inevitably be a trade-off between what you *can* store and what will be cost-effective to store.
3 - In practice any metadata you store onchain should meet some *descriptive standard* (for interoperability etc.). However these metadata standards for NFTs (across all blockchains) are very much still in discussion/development. Therefore in the short-term there is an onus on our DAO to come up with a standard (or use an existing one).
4 - It is likely (in my view) that only key NFT metadata will be stored onchain with a lot of the digital product being stored offchain in distributed locations like IPFS. 
5 - Typically any data would be hashed (cryptographically rolled up) onchain. Though there will be cases where you want open data onchain (so users can show off their purchase etc.).

In short ... you can store anything you want onchain subject to the size limitations, associated transaction charges and some descriptive standard accessible to users.
